sdks/python/apache_beam/io/gcp/bigtableio.py

```diff
@@ -122,22 +131,147 @@ class WriteToBigTable(beam.PTransform):
+      instance_id: [str] GCP Instance to write the Rows
+      table_id: [str] GCP Table to write the `DirectRows`
+      filter_: [RowFilter] Filter to apply to columns in a row.
+    """
+    super(self.__class__, self).__init__()
+    self._initialize({'project_id': project_id,
+                      'instance_id': instance_id,
+                      'table_id': table_id,
+                      'start_key': start_key,
+                      'end_key': end_key,
+                      'filter_': filter_})
+
+  def __getstate__(self):
+    return self._beam_options
+
+  def __setstate__(self, options):
+    self._initialize(options)
+
+  def _initialize(self, options):
+    self._beam_options = options
+    self.table = None
+    self.sample_row_keys = None
+    self.row_count = Metrics.counter(self.__class__.__name__, 'Rows read')
+
+  def start_bundle(self):
+    if self.table is None:
+      self.table = Client(project=self._beam_options['project_id'])\
+                    .instance(self._beam_options['instance_id'])\
+                    .table(self._beam_options['table_id'])
+
+  # def process(self, element, tracker=beam.DoFn.RestrictionTrackerParam):
+  def process(self, element, **kwargs):
+    for row in self.table.read_rows(start_key=self._beam_options['start_key'],
+                                    end_key=self._beam_options['end_key'],
+                                    filter_=self._beam_options['filter_']):
+      self.written.inc()
+      yield row
+
+  def get_initial_restriction(self, element):
+    pass
+
+  def finish_bundle(self):
+      pass
+
+  def display_data(self):
+    return {'projectId': DisplayDataItem(self._beam_options['project_id'],
+                                         label='Bigtable Project Id'),
+            'instanceId': DisplayDataItem(self._beam_options['instance_id'],
+                                          label='Bigtable Instance Id'),
+            'tableId': DisplayDataItem(self._beam_options['table_id'],
+                                       label='Bigtable Table Id'),
+            'filter_': DisplayDataItem(self._beam_options['filter_'],
+                                       label='Bigtable Filter',
+                                       key='filter_')
+            }
+
+
+class ReadFromBigTable(beam.PTransform):
+  def __init__(self, project_id, instance_id, table_id, filter_=None):
+    """ The PTransform to access the Bigtable Read connector
+
+    Args:
+      project_id: [str] GCP Project of to read the Rows
+      instance_id): [str] GCP Instance to read the Rows
+      table_id): [str] GCP Table to read the Rows
+      filter_: [RowFilter] Filter to apply to columns in a row.
+    """
+    super(self.__class__, self).__init__()
+    self._beam_options = {'project_id': project_id,
+                         'instance_id': instance_id,
+                         'table_id': table_id,
+                         'filter_': filter_}
+    self.table = None
```

Comment:
You shouldn't need class variables for `table` and `sample_row_keys`.  That should help with pickling